### PR TITLE
Move calls to asyncore to the loop thread

### PR DESCRIPTION
**Pull request by bjmb**
We should remove the changes from the build.yaml before merging this PR.

### CODE REVIEW

File: cassandra/io/asyncorereactor.py [hunk 1]
```diff
@@ -239,6 +267,11 @@ def _run_loop(self):
     def add_timer(self, timer):
         self._timers.add_timer(timer)
 
+        # This function is called from a different thread than the
+        # event loop thread, so for this call is thread safe, we must wake up
```
Comment (mambocab) [inline cassandra/io/asyncorereactor.py]: This phrasing confused me at first -- I expected to find a data race or something.

I think I used to understand `wake_loop` enough to rephrase, but I don't anymore :grimacing:

**Comment by mambocab** [comment]
So, explaining each commit:


- `whitespace`
  - Just a little cleanup for readability, and adding back a file-terminating newline that shouldn't have been removed. Kinda surprised your editor allows this.
- `rename Handle->WaitableTimer`
  - `Handle` makes it sound like this is a file handle or something. I think `WaitableTimer` captures the purpose of this class -- it's a wrapper around `Timer` with a `.wait` method.
- `grammar correction` and `rewrap comment` -- self-explanatory
- `change decorator name`
  - `notbigclockgranularity` is hard to parse -- eventually it makes sense, "this test should _not_ run if the clock granularity is big", but it took me a minute. I don't know if I like the name I landed on; `skipIfLargeClockGranularity` might be more in keeping with `unittest` conventions.

File: cassandra/io/asyncorereactor.py [hunk 2]
```diff
@@ -295,6 +328,10 @@ def handle_fork(cls):
     @classmethod
     def create_timer(cls, timeout, callback):
         timer = Timer(timeout, callback)
+        return cls.add_timer(timer)
+
+    @classmethod
+    def add_timer(cls, timer):
```
Comment (bjmb) [inline cassandra/io/asyncorereactor.py]: Yep, makes sense, I had doubts about adding it.
